### ISSUE TITLE
[Feature request] Ignore nodes in pathfinding

### ISSUE THREAD

**Issue by rkfg** [lightningnetwork/lnd#5488]
There are some popular nodes in the network that have great capacity and many channels but those channels can't route payments. Every time I see any of them in the path while rebalancing my own channels I can easily predict the payment will fail on their side and I'm almost always correct. I suppose they simply don't have enough outbound capacity, people open channels to them but they themselves don't do that. And even if they have a couple of channels with outbound capacity it's very improbable to find them among almost a thousand of other inbound-mostly channels. And they use minimal fees instead of hinting at the channels with sats in them.

Some examples of such nodes:
- [gameb_1](https://terminal.lightning.engineering/#/023d70f2f76d283c6c4e58109ee3a2816eb9d8feb40b23d62469060a2b2867b77f)
- [gameb_2](https://terminal.lightning.engineering/#/03c8dfbf829eaeb0b6dab099d87fdf7f8faceb0c1b935cd243e8c1fb5af71361cf)
- [⚡️ bCyber ⚡️](https://terminal.lightning.engineering/#/026db2cbf3d8ab4a4c01eed2df432d20cf0a13136402097574209d2595cb9e9d93)

It's very fascinating that the terminal was able to figure out their uselessness despite the high capacity/channel numbers because this correlates with my experience with those nodes. My autopilot opened channels to them and after several days of very tedious attempts to rebalance I closed them. However, I'd like to blacklist them completely so that I won't waste time on them anymore. I didn't find anything like that in the config and issues so asking here.

**Comment by tiker** [comment]
Related question:
Is there a way to tell if a remote LND node has "rejecthtlc=true" set to prevent routing transactions before opening a channel with it?  It is most likely set for the gameb nodes.

Knowing that would certainly help people and the autopilot scripts avoid opening channels with remote nodes that won't route payments.

**Comment by rkfg** [comment]
I think those nodes still route payments sometimes, during rebalancing I remember having failures on the channels _after_ them. They just seriously lack the capacity to do so.

**Comment by Roasbeef** [comment]
> However, I'd like to blacklist them completely so that I won't waste time on them anymore.

Yeah this is definitely possible. Right now in the `SendPayment` request, we have fields to ensure that we use a certain hop when sending the payment out, or a final hop (for rebalancing). The change here would be to add a list of edges that should just always be ignored during path funding.

**Comment by Roasbeef** [comment]
Actually today [you can use the `QueryRoutes`](https://api.lightning.community/#queryroutes) call to obtain a set of routes ignoring a set of peers by setting the `ignored_edges` and `ignored_nodes` fields. Underneath, this ends up hitting the same code as normal send, payment, so what needs to be done is adding those ignored fields to the normal send payment request.

**Comment by ErikEk** [comment]
Related to https://github.com/lightningnetwork/lnd/issues/4959
